src/Symfony/Component/Translation/MessageCache.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Component\Config\ConfigCache;
+use Symfony\Component\Config\Resource\FileResource;
+
+/**
+ * @author Abdellatif Ait Boudad <a.aitboudad@gmail.com>
+ */
+class MessageCache implements MessageCacheInterface
+{
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
+    /**
+     * @var bool
+     */
+    private $debug;
+
+    /**
+     * @param string $cacheDir
+     * @param bool   $debug
+     */
+    public function __construct($cacheDir, $debug = false)
+    {
+        $this->cacheDir = $cacheDir;
+        $this->debug = $debug;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isFresh($locale, $options = array())
+    {
+        if (!isset($options['resources_hash'])) {
+            $options['resources_hash'] = '';
+        }
+
+        $cache = new ConfigCache($resource = $this->getResourcePath($locale), $this->debug);
+        $isFresh = $cache->isFresh();
+        if ($isFresh && $this->debug) {
+            list($catalogue, $resourcesHash) = include $resource;
+            if ($resourcesHash !== $options['resources_hash']) {
+                return false;
+            }
+        }
+
+        return $isFresh;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load($locale)
+    {
+        $resource = $this->getResourcePath($locale);
+
+        list($catalogue, $resourcesHash) = include $resource;
+        $catalogue->addResource(new FileResource($resource));
```

Comment:
adding the cache file as a resource looks weird to me. Are you sure about that ?